src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -283,35 +283,45 @@ void remove_redundant_reorders::run(program& p) {
+            i_layout.data_type == o_layout.data_type ) {
             // If the newly aligned pad is merged into output layout during post_optimize_graph phase
             // and then buffer is reinterpreted, user node cannot handle pad properly for kernel execution
-            if (!update_implementations || (i_layout.feature() % 16 == 0 &&
-                i_layout.data_padding == padding() && o_layout.data_padding == padding()) || i_layout.batch() == 1) {
-                r_node.can_be_optimized(true);
-                r_node.requires_reinterpret(true);
+            if (i_layout.format == format::b_fs_yx_fsv16 && o_layout.format == format::bfyx &&
+                !layout_optimizer::onednn_check_preferred_impl_type_of_users(r_node)) {
+                if (!update_implementations || (i_layout.feature() % 16 == 0 &&
+                    i_layout.data_padding == padding() && o_layout.data_padding == padding()) || i_layout.batch() == 1) {
+                    r_node.can_be_optimized(true);
+                    r_node.requires_reinterpret(true);
 
-                auto pad_lo = o_layout.data_padding.lower_size();
-                auto pad_hi = o_layout.data_padding.upper_size();
+                    auto pad_lo = o_layout.data_padding.lower_size();
+                    auto pad_hi = o_layout.data_padding.upper_size();
 
-                pad_lo.batch[0] = i_layout.data_padding.lower_size().batch[0];
-                pad_hi.batch[0] = i_layout.data_padding.upper_size().batch[0];
+                    pad_lo.batch[0] = i_layout.data_padding.lower_size().batch[0];
+                    pad_hi.batch[0] = i_layout.data_padding.upper_size().batch[0];
 
-                pad_lo.feature[0] = i_layout.data_padding.lower_size().feature[0];
-                pad_hi.feature[0] = i_layout.data_padding.upper_size().feature[0];
+                    pad_lo.feature[0] = i_layout.data_padding.lower_size().feature[0];
+                    pad_hi.feature[0] = i_layout.data_padding.upper_size().feature[0];
 
-                if (i_layout.feature() % 16 != 0) {
-                    pad_hi.feature[0] += 16 - i_layout.feature() % 16;
-                }
+                    if (i_layout.feature() % 16 != 0) {
+                        pad_hi.feature[0] += 16 - i_layout.feature() % 16;
+                    }
 
-                r_node.merge_output_padding(padding{pad_lo.sizes(), pad_hi.sizes()});
-                continue;
+                    r_node.merge_output_padding(padding{pad_lo.sizes(), pad_hi.sizes()});
+                    continue;
+                }
+            } else if (i_layout.format == format::bfyx && o_layout.format == format::b_fs_yx_fsv16) {
+                if ((o_layout.feature() % 16 == 0 &&
+                    i_layout.data_padding == padding() && o_layout.data_padding == padding())) {
+                    r_node.can_be_optimized(true);
+                    r_node.requires_reinterpret(true);
+                    continue;
+                }
```

Comment:
what about implementing it within `layout_optimizer::compatible`? It would be more natural if it does not require any additional pad handling.

Reply:
Basically agree. But isn't it too specific to implement the bfyx and fsv16 compatibility check in `layout_optimizer::compatible`?  I don't have idea how many formats should be supported there pattern in `layout_optimizer::compatible`.

Reply:
Applied it in `check_redundant_1d_along_feature`